src/chemenv/tools/pubchem.py

```
@@ -466,260 +557,110 @@ async def _get_number_chiral_atoms(self) -> Optional[int]:
+
+                    # Handle other content types
                     else:
-                        results[orig_key] = None
-
-            counter += 1
-
-        return results
+                        # Try JSON first as a fallback
+                        try:
+                            record_data = await record_response.json()
+                            return record_data
```

Ruff (0.15.0)

[warning] Consider moving this statement to an `else` block

(TRY300)